src/states/node.rs

```diff
@@ -771,18 +774,28 @@ impl Node {
             let _ = cached_events.insert(event);
         }
         let our_pfx = *self.chain.our_prefix();
-        // filter cached events to SectionInfo where we benefit from additional signatures for
-        // neighbours for sec-msg-relay. Online/Offline events we only need to re-vote events which
-        // havent yet accumulated in old prefix and that are relevant to our new prefix.
+
         cached_events
             .iter()
             .filter(|event| match **event {
+                // Only re-vote not yet accumulated events and still relevant to our new prefix.
```

Comment:
This comment (L781 and piece of functionality L783) are duplicated, so should probably go to a function.
Actually, a good name for that function can probably remove the need for the comment :wink:

Reply:
Since it would be a free function taking 4 parameters, I'd rather not.
I'm happy to add a closure though.

Reply:
Though if you have a name in mind for this, because I'm not sure I see an obvious one that is not like the sentence in the comment.

Reply:
Not convinced will get a win there.

Reply:
Your argument for a lambda is convincing: What about:
```
// Only revote not yet accumulated events that are still relevant to our new prefix.
let should_revote = | prefix: Prefix | { our_pfx.matches(prefix) && !completed_events.contains(event) };
...
    NetworkEvent::Online(pub_id, _) | NetworkEvent::Offline(pub_id) => {
        should_revote(pub_id.name())
    }
    NetworkEvent::ExpectCandidate(ExpectCandidatePayload { ref dst_name, .. }) => {
        should_revote(dst_name)
    }
```

Reply:
Is that within the filter closure, because otherwise you are also needing event.
That is why I'm, not convinced.

Also the name is somewhat problematic, because other variant decide whether to re-vote differently.